From 283b4bf1621542d2fe14e8492b7578bf9c083418 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Fri, 25 Aug 2017 13:06:50 +0200
Subject: [PATCH] hooks.c: Fix build with glibc-2.26

* add missing sys/uio.h include for writev
* replace cfree with free as the cfree man says:
  This  function  should  never  be used.  Use free(3) instead.
  Starting with version 2.26, it has been removed from glibc.
* fixes:
  hooks.c: At top level:
  hooks.c:2428:26: error: 'cfree' undeclared here (not in a function); did you mean 'free'?
       HOOK_DIRECT_NO_DEBUG(cfree),
                          ^
  hooks.c:2675:26: error: 'writev' undeclared here (not in a function); did you mean 'write'?
       HOOK_DIRECT_NO_DEBUG(writev),
                            ^

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 hybris/common/hooks.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/hybris/common/hooks.c b/hybris/common/hooks.c
index 8a9cb2a..8c0f4d2 100644
--- a/hybris/common/hooks.c
+++ b/hybris/common/hooks.c
@@ -64,6 +64,7 @@
 #include <sys/syscall.h>
 #include <sys/auxv.h>
 #include <sys/prctl.h>
+#include <sys/uio.h>
 
 #include <sys/mman.h>
 #include <libgen.h>
@@ -2425,7 +2426,7 @@ static struct _hook hooks_common[] = {
     HOOK_DIRECT(malloc),
     HOOK_DIRECT_NO_DEBUG(free),
     HOOK_DIRECT_NO_DEBUG(calloc),
-    HOOK_DIRECT_NO_DEBUG(cfree),
+    HOOK_DIRECT_NO_DEBUG(free),
     HOOK_DIRECT_NO_DEBUG(realloc),
     HOOK_DIRECT_NO_DEBUG(memalign),
     HOOK_DIRECT_NO_DEBUG(valloc),
-- 
2.14.1

